src/charm.py

```diff
@@ -163,63 +134,85 @@ def _on_install(self, _: InstallEvent) -> None:
+        self._set_peer_data("started", False)
+        self.unit.status = BlockedStatus("Unit is down, service has stopped")
 
     def _on_config_changed(self, _: ConfigChangedEvent) -> None:
         """Triggered on config changed event."""
-        self._stored.bind_addr = self._get_bind_address()
         self._configure_srsenb_service()
-        if self._stored.started:
+        if self._get_peer_data("started"):
             self.unit.status = MaintenanceStatus("Reloading srsenb")
             service_restart(SRS_ENB_SERVICE)
-        self.unit.status = self._get_current_status()
+        self.unit.status = ActiveStatus(self._active_status_msg)
 
     def _on_update_status(self, _: UpdateStatusEvent) -> None:
         """Triggered on update status event."""
-        self.unit.status = self._get_current_status()
+        self.unit.status = ActiveStatus(self._active_status_msg)
+
+    def _on_lte_core_available(self, event: LTECoreAvailableEvent) -> None:
+        """Triggered on lte_core_available.
+
+        Retrieves MME address from relation, configures the srs enb service and restarts it.
+        """
+        if not self.unit.is_leader():
+            return
+        self._set_peer_data("mme_addr", event.mme_ipv4_address)
+        logging.info(f"MME IPv4 address from LTE core: {event.mme_ipv4_address}")
+        self._configure_srsenb_service()
+        if self._get_peer_data("started"):
+            self.unit.status = MaintenanceStatus("Reloading srsenb.")
+            service_restart(SRS_ENB_SERVICE)
+            logging.info("Restarting EnodeB after MME IP address change.")
+        self.unit.status = ActiveStatus(self._active_status_msg)
 
     def _on_attach_ue_action(self, event: ActionEvent) -> None:
         """Triggered on attach_ue action."""
-        self._stored.ue_usim_imsi = event.params["usim-imsi"]
-        self._stored.ue_usim_k = event.params["usim-k"]
-        self._stored.ue_usim_opc = event.params["usim-opc"]
+        if not self.unit.is_leader():
+            return
+        self._set_peer_data("ue_usim_imsi", event.params["usim-imsi"])
+        self._set_peer_data("ue_usim_k", event.params["usim-k"])
+        self._set_peer_data("ue_usim_opc", event.params["usim-opc"])
         self._configure_srsue_service()
         service_restart(SRS_UE_SERVICE)
-        self._stored.ue_attached = True
-        self.unit.status = self._get_current_status()
+        self._set_peer_data("ue_attached", True)
+        self.unit.status = ActiveStatus(self._active_status_msg)
         event.set_results({"status": "ok", "message": "Attached successfully"})
 
     def _on_detach_ue_action(self, event: ActionEvent) -> None:
         """Triggered on detach_ue action."""
-        self._stored.ue_usim_imsi = None
-        self._stored.ue_usim_k = None
-        self._stored.ue_usim_opc = None
+        if not self.unit.is_leader():
+            return
+        self._set_peer_data("ue_usim_imsi", None)
```

Comment:
Same here, should we check for the peer relation to be joined?

Reply:
not using peer data anymore :)